lib/Views/LauncherAuth.jsx

```diff
@@ -0,0 +1,162 @@
+        Authorization: authorization
+      }
+    })
+      .then(response => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        return response;
+      })
+      .then(response => response.json());
+
+    urlPromise.then(
+      response => {
+        localStorage.setItem("ewatercycle.launcher.token", response.token);
+        this.setState({
+          username: "",
+          password: ""
+        });
+      },
+      reason => {
+        this.props.viewState.terria.raiseErrorToUser(
+          reason,
+          "Login failed. Please try again"
+        );
+        console.error("bad request in experiment launcher: " + reason);
+      }
+    );
+  }
+
+  logOut() {
+    localStorage.removeItem("ewatercycle.launcher.token");
+    this.setState({
+      username: "",
+      password: ""
+    });
+  }
+
+  isAuthenticated() {
+    return getPersistentItem("ewatercycle.launcher.token");
+  }
+
+  render() {
+    const dropdownTheme = {
+      inner: Styles.dropdownInner
+    };
+    // TODO add icon like https://octicons.github.com/icon/sign-in/
+
+    if (this.isAuthenticated()) {
+      return (
+        <div className="tjs-panel__panel">
+          <button
+            onClick={this.logOut}
+            type="button"
+            className="tjs-panel__button tjs-_buttons__btn tjs-_buttons__btn--map"
+            title="Logout"
+          >
+            <span>Logout</span>
+          </button>
+        </div>
+      );
+    }
+    const jupyterHubUrl = "/jupyter";
+    return (
+      <MenuPanel
+        theme={dropdownTheme}
+        btnText="Login"
+        smallScreen={this.props.smallScreen}
+        viewState={this.props.viewState}
+        btnTitle="Login to start experiments"
+      >
+        <div className={classNames(PanelStyles.header)}>
+          <label className={PanelStyles.heading}>
+            Login to start experiments. Same credentials as JupyterHub
+          </label>
+        </div>
+        <div className={classNames(PanelStyles.section, Styles.section)}>
+          <form onSubmit={this.onSubmit}>
+            <label>
+              Username
+              <input
+                className="tjs-_form__field tjs-_form__input"
+                type="text"
+                value={this.state.username}
+                onChange={this.onUsernameChange}
+              ></input>
+            </label>
+            <label>
+              Password
+              <input
+                className="tjs-_form__field tjs-_form__input"
+                type="password"
+                value={this.state.password}
+                onChange={this.onPasswordChange}
+              ></input>
+            </label>
+            <p>
+              Make sure you logged into{" "}
+              <a href={jupyterHubUrl} target="_blank" className={Styles.link}>
+                Jupyter Hub
+              </a>{" "}
+              at least once with supplied username.
```

Comment:
You mean: the launcher doesn't work if you have never (directly) logged into Jupyter before?

Reply:
yep, the first login prepares JH for access using API.